packages/sit-onyx/src/i18n/index.ts

```diff
@@ -0,0 +1,135 @@
+import type { DeepPartial, ObjectToDottedStrings, TranslationValue } from "@/types";
+import { computed, inject, provide, unref, type InjectionKey, type MaybeRef } from "vue";
+import enUS from "./locales/en-US.json";
+
+/** Available translations that are used by Onyx components. */
+export type Translation = typeof enUS;
+
+export type ProvideI18nOptions = {
+  /**
+   * Current locale / language to use.
+   * If a ref is passed (e.g. the locale from the `vue-i18n` package)
+   * all Onyx messages will be updated if it changes (if locale is supported).
+   * If a message is missing for your currently set locale, English will be used as fallback.
+   *
+   * @default "en-US"
+   */
+  locale?: MaybeRef<string>;
+  /**
+   * Available translations / messages. English is always supported. For build-in translations, see:
+   * https://onyx.schwarz/i18n/
+   *
+   * @example
+   * ```ts
+   * import deDE from "sit-onyx/locales/de-DE.json";
+   * {
+   *   messages: {
+   *     // English is always supported so we don't need to add it here
+   *     'de-DE': deDE
+   *   }
+   * }
+   * ```
+   */
+  messages?: Record<string, DeepPartial<Translation> | Record<string, TranslationValue>>;
+};
+
+const I18N_INJECTION_KEY = Symbol() as InjectionKey<ReturnType<typeof createI18n>>;
+
+/**
+ * Creates a new i18n instance.
+ */
+const createI18n = (options?: ProvideI18nOptions) => {
+  /**
+   * Current locale.
+   * @default "en-US"
+   */
+  const locale = computed(() => unref(options?.locale) ?? "en-US");
+
+  const messages = computed(() => {
+    if (options?.messages && locale.value in options.messages) {
+      return options.messages[locale.value];
+    }
+    return enUS;
+  });
+
+  /**
+   * Gets the translation for the given key.
+   * If message is not found for current locale, English fallback will be used.
+   * @param placeholders Placeholders that will be replaced in the message string
+   * For pluralization, you must provide the placeholder `n`.
+   */
+  const t = computed(() => {
+    return (
+      key: ObjectToDottedStrings<Translation>,
+      placeholders: Record<string, string | number | undefined> = {},
+    ): string => {
+      // use English message as fallback
+      let message = resolveMessage(key, messages.value) ?? resolveMessage(key, enUS) ?? "";
+
+      if (placeholders) {
+        Object.entries(placeholders).forEach(([key, value]) => {
+          if (value === undefined) return;
+          // "gi" is used to replace all occurrences because String.replaceAll() is not available
```

Comment:
I don't understand
![image](https://github.com/SchwarzIT/onyx/assets/151019977/24e2c7a2-6d4e-48a3-90c1-58746a03fe4e)

but inside `packages/sit-onyx/tsconfig.vitest.json` ES2021 is already specified?
![image](https://github.com/SchwarzIT/onyx/assets/151019977/dc2e3681-e376-4856-8af4-6bc77393bb05)

Reply:
I mean your solution is also totally fine, I just don't get why this setting seems to be ignored

Reply:
Because the `tsconfig.vitest.json` is used for tests/type checking. For the actual build/component library the `tsconfig.app.json` is used :) It was probably a leftover, I removed it

Reply:
aah, right 🤦🏻‍♀️